docs/allocation.md

```diff
@@ -0,0 +1,152 @@
+
+### Allocation caches
+
+To improve performance, allocation caches are reserved in the heap for different threads. These allocation caches are known as thread local heaps (TLH) and allow each thread to allocate memory from its cache without acquiring the heap lock. A TLH is typically used for small objects of less than 512 bytes (768 bytes on 64-bit VMs) although larger objects can be allocated from the cache if there is sufficient space.
+
+If a thread allocates a lot of objects, the allocator gives that thread a larger TLH to reduce contention on the heap lock.
+
+A TLH is predefined with an initial default size of 2 KB. On every TLH refresh, the requested size for that thread
+is increased by an increment (default 4 KB). The requested size can grow up to a predefined maximum (default 128 KB).
+
+After every garbage collection cycle, the TLH requested size for each thread is reduced, sometimes by as much as 50%, to take account of threads that
+reduce their allocation rate and no longer need large TLHs.
+
+For very inactive threads, the requested size can even drop below the initial value, down to the predefined minimum (512/768 bytes).
+For very active threads, the maximum TLH requested size might be reached before the next GC occurs.
+
+Larger TLHs can help reduce heap lock contention, but might also reduce heap utilization and increase heap fragmentation.
+
+The following options control the requested TLH size:
+
+- [`-Xgc:tlhMaximumSize=<bytes>`](xgc.md#tlhmaximumsize)
+- [`-Xgc:tlhInitialSize=<bytes>`](xgc.md#tlhinitialsize)
+- [`-Xgc:tlhIncrementSize=<bytes>`](xgc.md#tlhincrementsize)
+
+Typically, when the maximum TLH size is increased, you should also increase the increment proportionally, so that active threads can
+reach the maximum requested TLH size more quickly.
+
+## Heap configuration
+
+Depending on the memory management strategy in force, the Java heap can be configured in a number of ways. The simplest configuration consists of a single area of memory, often referred to as a *flat* heap. Other configurations divide the heap into different *areas* or *regions*, which might contain objects of different ages (generations) or sizes.
+
+### Area-based heaps
+
+The default GC policy for OpenJ9 uses a heap configuration that is divided into two main areas: the *nursery* area for new object allocation, and the *tenure* area for objects that continue to be reachable for a longer period of time. Most objects have short lifecycles and can be reclaimed by the garbage collector more quickly by focusing only on the nursery area. Global GC cycles that cause application pauses in order to clear and defragment the *tenure* area are less frequent.
+
+### SOA and LOA
+
+Some GC policies subdivide a *flat* heap or an *area* of the heap for object allocation into the Small Object Area (SOA) and the Large Object Area (LOA).
```

Comment:
Actually, but default, all (not some) area based heaps will have LOA.

Reply:
Changed to "All area-based heaps subdivide part of the heap into the Small Object Area (SOA) and the Large Object Area (LOA)."